WordSearch.java

```diff
@@ -0,0 +1,42 @@
+import java.util.ArrayList;
+import java.util.List;
+
+public class WordSearch {
+    private List<String> results = new ArrayList<>();
+
+    public void findAllWords(int col, int row, char[][] characters, Dictionary dictionary) {

```

Comment:
It would be good to have more descriptive names for parameters (plus some comments...). `col` and `row` suggest specific row and column rather than number of rows/columns.